src/video/SDL_surface.c

```diff
@@ -139,19 +154,36 @@ SDL_Surface *SDL_CreateSurface(int width, int height, SDL_PixelFormatEnum format
         return NULL;
     }
 
-    surface->format = SDL_CreatePixelFormat(format);
-    if (!surface->format) {
-        SDL_DestroySurface(surface);
-        return NULL;
-    }
+    surface->format = format;
     surface->w = width;
     surface->h = height;
     surface->pitch = (int)pitch;
-    SDL_SetSurfaceClipRect(surface, NULL);
 
-    if (SDL_ISPIXELFORMAT_INDEXED(surface->format->format)) {
-        SDL_Palette *palette =
-            SDL_CreatePalette((1 << surface->format->bits_per_pixel));
+    surface->internal = (SDL_SurfaceData *)SDL_calloc(1, sizeof(*surface->internal));
```

Comment:
More of a future TODO, but it would be nice to fuse the surface and surfacedata allocations. Basically do what SDL_CreateStackSurface already does.

Reply:
Yeah, I think that's a good idea.

Reply:
Done, this was a good change, thanks!
Check out SDL_InitializeSurface() for details.